ner.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/python3
+
+import argparse
+import csv
+from glob import glob
+from os.path import isdir, basename
+from itertools import groupby, zip_longest
+import spacy
+
+
+def main():
+    args = parse_args()
+
+    files = []
+
+    if args.input_dir:
+        files = glob(f"{args.input_dir}/*.txt")
+
+    if args.file:
+        for f in args.file:
+            files.append(f)
+
+    entity_dicts = parse_ner(files)
+    write_entities(entity_dicts, args.output_dir)
+
+
+def parse_args() -> dict:
+    parser = argparse.ArgumentParser(
+        description="A script to scan text and output NER information"
+    )
+    parser.add_argument("-i", "--input_dir", type=dir_path, nargs="?")
+    parser.add_argument("-o", "--output_dir", type=dir_path, nargs="?", default=".")
+    parser.add_argument("-f", "--file", nargs="+")
+
+    return parser.parse_args()
+
+
+def dir_path(path: str) -> str:
+    if isdir(path):
+        return path
+    else:
+        raise argparse.ArgumentTypeError(f"{path} is not a valid directory")
+
+
+def parse_ner(files: list) -> dict:
+    nlp = spacy.load("en_core_web_sm")
+
+    entity_dicts = {}
+
+    for file in files:
+        with open(file) as f:
+            file_root = basename(file).rpartition("-")[0]
+            doc = nlp(f.read())
+
+            entities = {
```

Comment:
Please add comments to document this dictionary comprehension.  It might be clearer if moved to its own named method (along with copious comments).  Show a small sample of output or otherwise document the resulting dictionary.